src/Processors/Formats/Impl/Parquet/ParquetBloomFilterCondition.cpp

```diff
@@ -0,0 +1,295 @@
+#include <Processors/Formats/Impl/Parquet/ParquetBloomFilterCondition.h>
+
+#if USE_PARQUET
+
+#include <parquet/bloom_filter.h>
+
+#include <DataTypes/DataTypeTuple.h>
+#include <DataTypes/DataTypeArray.h>
+#include <DataTypes/DataTypeNullable.h>
+#include <DataTypes/DataTypeLowCardinality.h>
+#include <Interpreters/misc.h>
+#include <Interpreters/convertFieldToType.h>
+#include <Columns/ColumnConst.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int LOGICAL_ERROR;
+    extern const int BAD_ARGUMENTS;
+}
+
+namespace
+{
+    parquet::ByteArray createByteArray(std::string_view view, TypeIndex type, uint8_t * buffer, uint32_t buffer_size)
+    {
+        if (isStringOrFixedString(type))
+        {
+            return view;
+        }
+        else
+        {
+            auto size = static_cast<uint32_t>(std::max(view.size(), sizeof(uint32_t)));
+            chassert(size <= buffer_size);
+            std::copy(view.begin(), view.end(), buffer);
```

Comment:
Wait, doesn't it need to be sign-extended when converting from [U]Int{8,16} to Int32? Test with something like:
```
insert into function file('t.parquet') select 150::UInt8 as u8, -30::Int8 as i8, 50000::UInt16 as u16, -10000::Int16 as i16 settings output_format_parquet_use_custom_encoder=0;
select u8 from file('t.parquet') where indexHint(u8 == 150);
select i8 from file('t.parquet') where indexHint(i8 == -30);
select u16 from file('t.parquet') where indexHint(u16 == 50000);
select i16 from file('t.parquet') where indexHint(i16 == -10000);
```
(And somehow check if bloom filter was actually used in the query. I'd step through it in debugger, or add debug logging, or add a `ProfileEvents` for how many times bf was used.)

My guess is that this code will have to know whether the parquet logical type is signed or unsigned, and sign-extend iff it's signed (which is also the default if logical/converted type is unset).